File: HackerRank/Problem_Solving/js/diagonalDifference.js
function diagonalDifference(arr) {
    // Write your code here
  const len = arr.length - 1;
  let diagPrincipal = 0;
  let diagSec = 0;
  for(let i=0; i <= len; i++){
    diagPrincipal+= arr[i][i];
  }

  for (let i = len; i >= 0 ; i --) {
    diagSec += arr[len-i][i] ;
  }

  return Math.abs(diagPrincipal - diagSec);
}

const matrix = [
  [1,2,3],
  [4,5,6],
  [0,8,5]];
console.log(diagonalDifference(matrix));

File: HackerRank/Problem_Solving/py/day2/diagonal_difference.py
def diagonalDifference(arr):
		# Write your code here
		diag1 = 0
		diag2 = 0
		for i in range(len(arr)):
				diag1 += arr[i][i]
				diag2 += arr[i][len(arr) - 1 - i]
		return abs(diag1 - diag2)


matrix = [
  [1,2,3],
  [4,5,6],
  [0,8,5]
]

print(diagonalDifference(matrix))
